app/controllers/api/v1/release_engines/npm/package_metadata_controller.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module Api::V1::ReleaseEngines
+  class Npm::PackageMetadataController < Api::V1::BaseController
+    before_action :scope_to_current_account!
+    before_action :require_active_subscription!
+    before_action :authenticate_with_token
+    before_action :set_package
+
+    def show
+      authorize! package,
+        to: :show?
+
+      artifacts = authorized_scope(package.artifacts.tarballs).order_by_version
+        .where_assoc_exists(:manifest) # must exist
+        .preload(:manifest,
+          release: %i[product entitlements constraints],
+        )
+      authorize! artifacts,
+        to: :index?
+
+      # FIXME(ezekg) https://github.com/brianhempel/active_record_union/issues/35
+      last_modified = artifacts.maximum(:"#{artifacts.table_name}.updated_at")
+      latest        = artifacts.first
+      metadata      = artifacts.reduce(
+        name: package.key,
+        time: { created: package.created_at, modified: last_modified },
+        'dist-tags': { latest: latest.version },
+        versions: {},
+      ) do |metadata, artifact|
+        package_json = artifact.manifest.as_package_json
+
+        metadata[:time][artifact.version]     = artifact.created_at.iso8601(3)
+        metadata[:'dist-tags'][artifact.tag]  = artifact.version if artifact.tag?
+        metadata[:versions][artifact.version] = package_json.merge(
+          dist: {
+            tarball: vanity_v1_account_release_artifact_url(current_account, artifact, filename: artifact.filename, host: request.host),
+            # FIXME(ezekg) only include valid SHA1 checksums?
+            shasum: artifact.checksum,
```

Comment:
Looks like there's also a SHA512 [`dist.integrity`](https://github.com/npm/cli/blob/780afc50e3a345feb1871a28e33fa48235bc3bd5/workspaces/libnpmpublish/lib/publish.js#L112) property? May need to address FIXME.

Reply:
Let's build an integrity hash separately, based on checksum format and length, and splat it into here.

Reply:
As far as I can tell, `integrity` follows the [SRI](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity) spec, i.e. checksums are base64 encoded.